### PR TITLE
Add DHCP agent information to data model

### PR DESCRIPTION
**Pull request by neiljerram**

### CODE REVIEW

**Comment by neiljerram** [comment]
Hi @fasaxc - The associated networking-calico change feels close enough now that it's worth looking at getting this data model change into the core Calico repository.  Please could you review it?

File: calico/datamodel_v1.py [hunk 1]
```diff
@@ -85,6 +86,10 @@
 ENDPOINT_STATUS_DOWN = "down"
 ENDPOINT_STATUS_ERROR = "error"
 
+# Information intended for use by the DHCP agent.
+DHCP_DIR = ROOT_DIR + "/dhcp" + DHCP_VERSION
+SUBNET_DIR= DHCP_DIR + "/subnet"
```
Comment (fasaxc) [inline calico/datamodel_v1.py]: Does it make sense to go straight to subnet rather than having an intermediate network directory?

Comment (fasaxc) [inline calico/datamodel_v1.py]: I'm thinking that we may, later, want to have some config attached to the whole OpenStack-style network and this prevents that

Comment (neiljerram) [inline calico/datamodel_v1.py]: Well I think so, from a DHCP perspective, for the same reason as commented at lines 121-128 of https://review.openstack.org/#/c/241310/17/networking_calico/agent/dhcp_agent.py.

Comment (neiljerram) [inline calico/datamodel_v1.py]: But would such config go under the 'dhcp' subtree, with 'dhcp' being explicitly specified as information for the DHCP agent?

Comment (fasaxc) [inline calico/datamodel_v1.py]: Not right now, but I'm wondering if we could get caught out later.  The only thing that I can see that we're concretely missing is the admin status for the network object.  OpenStack seems to leave everything else up to the subnet object.

**Pull request merged**